WeakAuras/SubRegionTypes/Tick.lua

```diff
@@ -0,0 +1,182 @@
+if not WeakAuras.IsCorrectVersion() then return end
+
+local SharedMedia = LibStub("LibSharedMedia-3.0");
+local L = WeakAuras.L;
+
+local default = function(parentType)
+  return {
+    tick_visible = true,
+    tick_color = {1, 1, 1, 1},
+    tick_placement_mode = "STATIC",
+    tick_placement = "50%",
+    automatic_height = true,
+    tick_width = 2,
+    tick_height = 30,
+    --tick_hide_mode = "NEVER",
+  }
+end
+
+local properties = {
+  tick_visible = {
+    display = L["Visibility"],
+    setter = "SetVisible",
+    type = "bool",
+    defaultProperty = true
+  },
+  tick_color = {
+    display = L["Color"],
+    setter = "SetTickColor",
+    type = "color"
+  },
+  tick_placement = {
+    display = L["Placement"],
+    setter = "SetTickPlacement",
+    type = "input",
+    validate = WeakAuras.ValidateNumericOrPercent,
+  },
+  tick_width = {
+    display = L["Width"],
+    setter = "SetTickWidth",
+    type = "number",
+    min = 0,
+    bigStep = 1,
+    default = 2,
+  },
+  tick_height = {
+    display = L["Height"],
+    setter = "SetTickHeight",
+    type = "number",
+    min = 0,
+    bigStep = 1,
+    default = 30,
+    --hidden = function() return data.automatic_height end,
+  },
+}
+
+local auraBarTrueLeft = {
+  ["HORIZONTAL"] = "LEFT",
+  ["HORIZONTAL_INVERSE"] = "RIGHT",
+  ["VERTICAL"] = "TOP",
+  ["VERTICAL_INVERSE"] = "BOTTOM",
+}
+
+local auraBarTrueLeftInverse = {
+  ["HORIZONTAL"] = "RIGHT",
+  ["HORIZONTAL_INVERSE"] = "LEFT",
+  ["VERTICAL"] = "BOTTOM",
+  ["VERTICAL_INVERSE"] = "TOP",
+}
+
+local function create()
+  local subRegion = CreateFrame("FRAME", nil, UIParent)
+  subRegion.texture = subRegion:CreateTexture()
+  subRegion.texture:SetDrawLayer("ARTWORK", 3)
+  subRegion.texture:SetAllPoints(subRegion)
+  return subRegion
+end
+
+local function onAcquire(subRegion)
+  subRegion:Show()
+end
+
+local function onRelease(subRegion)
+  subRegion:Hide()
+end
+
+local funcs = {
+  ["Update"] = function(self)
+    self:SetTickPlacement(self.tick_placement)
+  end,
+  SetVisible = function(self, visible)
+    if visible then
+      self:Show()
+    else
+      self:Hide()
+    end
+  end,
+  SetTickColor = function(self, r, g, b, a)
+    self.texture:SetColorTexture(r, g, b, a or 1)
+  end,
+  SetTickPlacement = function(self, placement)
+    local offset, offsetx, offsety = self.tick_placement, 0, 0
+    local width = self.parentTrueWidth
+    if self.tick_placement_mode == "STATIC" then
+      local percent = string.match(placement, "(%d+)%%")
+      if percent then
+        offset = (tonumber(percent) / 100) * width
+      else
+        local pixels = width / (self.parent.state.duration or self.parent.state.total or 1)
```

Comment:
This highlights a fairly substantial flaw, there is no max duration in situations such as a Cooldown Progress trigger that has not fired. This causes (intentionally through 101) it to stick to its last known location. If the trigger fires at least once it will be "fine" (assuming you [i]want[/i] the tick to remain there), but I'm not sure if it's even possible to handle situations where it hasn't fired. Having a default can look awkward in any situation.

Reply:
If there's no (or a infinite) duration, then imho ticks should be hidden in the STATIC/Absolute mode. So basically a tick is hidden if either it's explicitly or implicitly hidden.